force-app/main/default/classes/DMLOperations.cls

```diff
@@ -145,10 +178,49 @@ public with sharing class DMLOperations {
      * @param oppNames The list of Opportunity names.
      */
     public static void upsertOpportunities(String accountName, List<String> oppNames) {
-        // Loop through each Opportunity name
-            // Create an Opportunity with the given name
-            // Add the Opportunity to the list
-    }
+       Account myAccount = new Account (Name = accountName);
+       upsert myAccount;
+        
+ 
+         // Query the Account again to get the Id if it was inserted
+         myAccount = [SELECT Id FROM Account WHERE Name = :accountName WITH SECURITY_ENFORCED LIMIT 1];
+ 
+         // Create a list to hold the Opportunities
+         List <Opportunity> opportunitiesToUpsert = new List<Opportunity>();
+         
+         // query opportunies to check if they exist
+         Map<String, Opportunity> existingOpportunitiesMap = new Map<String, Opportunity>();
+         for (Opportunity opp : [SELECT Id, Name FROM Opportunity WHERE Name IN :oppNames AND AccountId = :myAccount.Id WITH SECURITY_ENFORCED]) {
+            existingOpportunitiesMap.put(opp.Name, opp);
+         }
+ 
+
+         // Loop through each Opportunity name
+        
+         for (String oppName : oppNames) {
+             // Create an Opportunity with the given name
+             Opportunity opp;
+             if (existingOpportunitiesMap.containsKey(oppName)){
+                opp = existingOpportunitiesMap.get(oppName);
+                opp.StageName = 'Qualification';
+                opp.CloseDate = Date.today().addMonths(1);
+                opp.Amount = 50000;
+             } else {
+                opp = new Opportunity(
+                    Name = oppName,
+                    AccountId = myAccount.Id,
+                    StageName = 'Qualification',
+                    CloseDate = Date.today().addMonths(1),
+                    Amount = 50000
+                );
+             }
+             opportunitiesToUpsert.add (opp); 
+         }
+
+         // Upsert the opportunities list so that records are either updated or created
+         upsert opportunitiesToUpsert;
```

Comment:
Great job checking to see if the opportunities already exist and upserting outside of your loop!